src/SUFST/src/can_unpack.c

```diff
@@ -3,25 +3,27 @@
 #include "can_handlers.h"
 #include "can_unpack.h"
 #include "usart.h"
+#include "rtcan.h"
+#include "can.h"
 
+#define RTCAN_THREAD_PRIORITY   3
 #define QUEUE_RX_THREAD_PRIORITY             10
 #define QUEUE_RX_THREAD_STACK_SIZE           1024
 #define QUEUE_RX_THREAD_PREEMPTION_THRESHOLD 10
 
 void queue_receive_thread_entry(ULONG input);
 
-UINT unpack_init(unpack_context_t* unpack_ptr, TX_BYTE_POOL* stack_pool_ptr, TX_QUEUE * can_queue){
+UINT unpack_init(unpack_context_t* unpack_ptr, TX_BYTE_POOL* stack_pool_ptr){
 
 VOID* thread_stack_ptr = NULL;
-unpack_ptr->rx_queue = can_queue;
-/* Setup transmit queue */
-    UINT tx_status = tx_queue_create(&unpack_ptr->tx_queue,
-                                     "SPI Data Receive Queue",
-                                     sizeof(pdu_t)/sizeof(ULONG),
-                                     &unpack_ptr->tx_queue_mem,
-                                     PDU_TX_QUEUE_SIZE * sizeof(pdu_t));
-
-    if(tx_status == TX_SUCCESS)
+
+    // initialise RTCAN instance
+    UINT tx_status = rtcan_init(&unpack_ptr->rtcan, 
+               &hcan1, 
```

Comment:
Just to double check, this CAN handle should be configured for 500kbps in CubeMX has that been done?

Reply:
What should be the exact value of can configuration. This is the current setting. 
![image](https://user-images.githubusercontent.com/47193610/222127369-6fd5ccaf-e749-42bf-8071-8a4eaad8b4b2.png)

Reply:
I think the setting is:
- 2 time quanta in bit segment 1
- 1 time quanta in bit segment 2
- Prescaler 18

You should then get a baud rate of 500,000. If not, check the IOC for the VCU or CAN test rig.

Reply:
Okay, I just realised the issue here.. The CAN relies on APB1 peripheral clock and this is input clock for prescaler. Since the value of this clock is different in vcu and can-test-rig, I need to use different value of presacaler in on-car-telemetry  For vcu the clock and prescaler is 36MHz and 18, whereas for on-car-telemetry it going to be 16MHz and 8. Just checked and the bitrate is indeed 500kbps now.

Reply:
Perfect, that makes sense. We can test the peripheral with actual hardware next Tuesday as well.